examples/tutorials/analysis-3d/event_sampling_nrg_depend_models.py

```diff
@@ -0,0 +1,413 @@
+"""
+Sample a source with energy-dependent temporal evolution
+==============
+This notebook shows how to sample events of sources whose model evolves in energy and time.
+
+Prerequisites
+-------------
+
+To understand how to generate a model and a MapDataset and how to fit the data, please refer to the `~gammapy.modeling.models.SkyModel` and
+:doc:`/tutorials/analysis-3d/simulate_3d` tutorial. To know how to sample events for standards sources, we suggest to visit the event sampler :doc:`/tutorials/analysis-3d/event_sampling` tutorial.
+
+Objective
+-------------
+
+Describe the process of sampling events of a source having an energy-dependent temporal model, and obtain an output event-list.
+
+Proposed approach
+-------------
+
+Here we will show how to create an energy dependent temporal model; then we also create an observation and define a Dataset object. Finally we describe how to sample events from the given model.
+
+We will work with the following functions and classes:
+
+-  `~gammapy.data.Observations`
+-  `~gammapy.datasets.Dataset`
+-  `~gammapy.modeling.models.SkyModel`
+-  `~gammapy.datasets.MapDatasetEventSampler`
+-  `~gammapy.data.EventList`
+-  `~gammapy.maps.RegionNDMap`
+"""
+
+######################################################################
+# Setup
+# -----
+#
+# As usual, let’s start with some general imports…
+#
+
+from pathlib import Path
+import numpy as np
+import astropy.units as u
+from astropy.coordinates import Angle, SkyCoord
+from astropy.time import Time
+from regions import CircleSkyRegion, PointSkyRegion
+import matplotlib.pyplot as plt
+from gammapy.data import Observation, observatory_locations
+from gammapy.datasets import MapDataset, MapDatasetEventSampler
+from gammapy.irf import load_irf_dict_from_file
+from gammapy.makers import MapDatasetMaker
+from gammapy.maps import Map, MapAxis, RegionNDMap, WcsGeom
+from gammapy.modeling import Fit
+from gammapy.modeling.models import (
+    ConstantSpectralModel,
+    FoVBackgroundModel,
+    LightCurveTemplateTemporalModel,
+    Models,
+    PointSpatialModel,
+    PowerLawSpectralModel,
+    SkyModel,
+)
+
+######################################################################
+# Check setup
+# -----------
+from gammapy.utils.check import check_tutorials_setup
+
+check_tutorials_setup()
+
+
+######################################################################
+# Create the energy-dependent model
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# The source to simulate has a spectrum that varies as a function of
+# the time.
+# We define its spatial morphology as `point-like`: we note that this
+# is a mandatory condition to simulate energy-dependent sources! Other,
+# extended morphologies will raise an error!
```

Comment:
Is there a way or motivation to support that in future ? I guess no, then maybe we can justify this design choice here ?